presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoDriverUri.java

```diff
@@ -103,12 +109,17 @@ private PrestoDriverUri(URI uri, Properties driverProperties)
     {
         this.uri = requireNonNull(uri, "uri is null");
         address = HostAndPort.fromParts(uri.getHost(), uri.getPort());
-        properties = mergeConnectionProperties(uri, driverProperties);
+        properties = mergeProvidedProperties(uri, driverProperties);
+
+        setSSLPropertyByPortIfNotProvided(properties, uri);
 
         validateConnectionProperties(properties);
 
+        setDefaultPropertiesIfNotProvided(properties);
+
         // enable SSL by default for standard port
-        useSecureConnection = SSL.getValue(properties).orElse(uri.getPort() == 443);
```

Comment:
you changed the logic here. Previously port could influence if SSL is enabled, now SSL is changing the port. Please revert that change.

Reply:
I move the logic to the following method. The logic should be the same.

Since we need to make `SSL_VERIFICATION` valid when `SSL` is not set by the user and the port is 443, the following method needs to be executed before validateConnectionProperties().

```
private static void setSSLPropertyByPortIfNotProvided(Properties properties, URI uri)
            throws SQLException
    {
        if (!SSL.getValue(properties).isPresent()) {
            properties.setProperty(SSL.getKey(), Boolean.toString((uri.getPort() == 443)));
        }
    }
```